header-verifier/tests/header_verifier_blackbox_test.rs

```diff
@@ -1,125 +1,313 @@
-#![allow(deprecated)]
-
 use bls_signature::BlsSignature;
-use header_verifier::ProxyTrait;
-use multiversx_sc::types::{ManagedBuffer, ManagedByteArray, MultiValueEncoded};
+use header_verifier::{header_verifier_proxy, Headerverifier};
+use multiversx_sc::{
+    api::ManagedTypeApi,
+    types::{
+        BigUint, ManagedBuffer, ManagedByteArray, MultiValueEncoded, TestAddress, TestSCAddress,
+    },
+};
 use multiversx_sc_scenario::{
-    api::StaticApi,
-    scenario_model::{Account, ScCallStep, ScDeployStep, SetStateStep, TxExpect},
-    ContractInfo, ScenarioWorld,
+    api::StaticApi, imports::MxscPath, multiversx_chain_vm::crypto_functions::sha256, DebugApi,
+    ExpectError, ScenarioTxRun, ScenarioTxWhitebox, ScenarioWorld,
 };
 
-const HEADER_VERIFIER_PATH_EXPR: &str = "file:output/header-verifier.wasm";
-const OWNER_ADDRESS_EXPR: &str = "address:owner";
-const LEADER_ADDRESS_EXPR: &str = "address:proposer";
-const VALIDATOR_ADDRESS_EXPR: &str = "address:board-member";
-const HEADER_VERIFIER_ADDRESS_EXPR: &str = "sc:header_verifier";
+const HEADER_VERIFIER_CODE_PATH: MxscPath = MxscPath::new("ouput/header-verifier.mxsc-json");
+const HEADER_VERIFIER_ADDRESS: TestSCAddress = TestSCAddress::new("header-verifier");
+
+// NOTE: This is a mock path
+const ENSHRINE_CODE_PATH: MxscPath =
+    MxscPath::new("../chain-factory/output/chain-factory.mxsc-json");
+const ENSHRINE_ADDRESS: TestSCAddress = TestSCAddress::new("enshrine");
+
+const OWNER: TestAddress = TestAddress::new("owner");
+const WEGLD_BALANCE: u128 = 100_000_000_000_000_000;
+
+type BlsKeys = MultiValueEncoded<StaticApi, ManagedBuffer<StaticApi>>;
 
-type HeaderVerifierContract = ContractInfo<header_verifier::Proxy<StaticApi>>;
+pub struct ErrorStatus<'a> {
```

Comment:
I think you can use ExpectError directly

Reply:
i am using this struct because it's easier to match it inside a propose function, one arm is when an error is expected and one for no error

Reply:
an alternative would be to use Option<ExpectError> or otherwise ExpectError(0, "") represents success

Reply:
each test that uses this struct has an Option<ExpectError>